nemo_curator/scripts/add_id.py

```diff
@@ -73,77 +69,25 @@ def attach_args(
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
     )
 ):
-    parser.add_argument(
-        "--input-data-dir",
-        type=str,
-        default=None,
-        help="Input directory consisting of .jsonl files that are accessible "
-        "to all nodes. Use this for a distributed file system",
-    )
-    parser.add_argument(
-        "--starting-index",
-        type=int,
-        default=None,
-        help="If supplied, determines the starting index from which to start "
-        "indexing the documents. By default, it is unspecified, and uses an id"
-        " scheme that is fast to calculate and is not guaranteed to be ordered.",
-    )
-    parser.add_argument(
-        "--output-data-dir",
-        type=str,
-        default=None,
-        help="The output directory to where the jsonl "
-        "files with ids will be written. If not specified, the ids will "
-        "be written in-place",
-    )
-    parser.add_argument(
-        "--id-field-name",
-        type=str,
-        default="adlr_id",
-        help="The name of the field that will contain the id value. "
-        "Default is 'adlr_id'",
-    )
-    parser.add_argument(
-        "--id-prefix",
-        type=str,
-        default="doc_id",
-        help="The prefix to the id number that will be assigned to the "
-        "document. When performing deduplication jointly with different"
-        "datasets, it is helpful to provide a prefix that denotes that a "
-        "document belongs to a particular dataset (e.g., wiki for documents"
-        "that come from the wikipedia dataset)",
-    )
-    attach_bool_arg(
-        parser,
-        "shuffle",
-        help_str="Shuffle the order of files before assigning IDs."
-        "Useful for creating a copy dataset with different IDs",
-    )
-    parser.add_argument(
-        "--seed",
-        type=int,
-        default=42,
-        help="If shuffling is specified, use this random seed to "
-        "perform the random shuffling",
-    )
-    parser.add_argument(
-        "--input-file-type",
-        type=str,
-        default="jsonl",
-        help="File type of the dataset to be read in. Supported file formats"
-        " include 'jsonl' (default), 'pickle', or 'parquet'.",
+    argumentHelper = ArgumentHelper(parser)
+
+    argumentHelper.add_id_field_name()
+    argumentHelper.add_id_prefix()
+    argumentHelper.add_input_data_dir()
+    argumentHelper.add_input_file_type()
+    argumentHelper.add_output_data_dir(
```

Comment:
Since this is so specific to `add_id` not sure if needs to be in the helper class. And if it does, I'd prefer grouping them all in a single `add_id_args` method rather than individual methods.

Reply:
Done!